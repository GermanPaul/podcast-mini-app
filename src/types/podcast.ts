export interface TopPodcasts {
  feed: Feed
}

export interface Feed {
  author: Author
  entry: Entry[]
  updated: Updated
  rights: FeedRights
  title: FeedTitle
  icon: Icon
  link: FeedLink[]
  id: FeedId
}

export interface Author {
  name: Name
  uri: Uri
}

export interface Name {
  label: string
}

export interface Uri {
  label: string
}

export interface Entry {
  'im:name': ImName
  'im:image': ImImage[]
  summary: Summary
  'im:price': ImPrice
  'im:contentType': ImContentType
  rights: Rights
  title: Title
  link: Link
  id: Id
  'im:artist': ImArtist
  category: Category
  'im:releaseDate': ImReleaseDate
}

export interface ImName {
  label: string
}

export interface ImImage {
  label: string
  attributes: ImImageAttributes
}

export interface ImImageAttributes {
  height: string
}

export interface Summary {
  label: string
}

export interface ImPrice {
  label: string
  attributes: PriceAttributes
}

export interface PriceAttributes {
  amount: string
  currency: string
}

export interface ImContentType {
  attributes: ContentTypeAttributes
}

export interface ContentTypeAttributes {
  term: string
  label: string
}

export interface Rights {
  label: string
}

export interface Title {
  label: string
}

export interface Link {
  attributes: LinkAttributes
}

export interface LinkAttributes {
  rel: string
  type: string
  href: string
}

export interface Id {
  label: string
  attributes: IdAttributes
}

export interface IdAttributes {
  'im:id': string
}

export interface ImArtist {
  label: string
  attributes: ArtistAttributes
}

export interface ArtistAttributes {
  href: string
}

export interface Category {
  attributes: CategoryAttributes
}

export interface CategoryAttributes {
  'im:id': string
  term: string
  scheme: string
  label: string
}

export interface ImReleaseDate {
  label: string
  attributes: ReleaseDateAttributes
}

export interface ReleaseDateAttributes {
  label: string
}

export interface Updated {
  label: string
}

export interface FeedRights {
  label: string
}

export interface FeedTitle {
  label: string
}

export interface Icon {
  label: string
}

export interface FeedLink {
  attributes: Attributes9
}

export interface Attributes9 {
  rel: string
  type?: string
  href: string
}

export interface FeedId {
  label: string
}
