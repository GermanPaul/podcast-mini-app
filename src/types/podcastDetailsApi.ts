export interface PodcastDetails {
  title: string
  author: string
  description: string
  image: string
  items: PodcastDetailsItem[]
}

export interface PodcastDetailsItem {
  id: string
  title: string
  date: string
  duration: string
  description: string
  url: string
}

export interface RawPodcastDetails {
  title: string;
  description: string;
  image?: { url: string };
  'itunes:author': string;
  'itunes:image': { href: string }
  item: RawPodcastDetailsItem[]
}

export interface RawPodcastDetailsItem {
  title: string;
  pubDate: string;
  'itunes:duration'?: string;
  description: string;
  enclosure: { url: string }
  guid: { '#text': string } | string
}

export interface PodcastDetailsApiResponse {
  resultCount: number
  results: Result[]
}

export interface Result {
  wrapperType: string
  kind: string
  artistId: number
  collectionId: number
  trackId: number
  artistName: string
  collectionName: string
  trackName: string
  collectionCensoredName: string
  trackCensoredName: string
  artistViewUrl: string
  collectionViewUrl: string
  feedUrl: string
  trackViewUrl: string
  artworkUrl30: string
  artworkUrl60: string
  artworkUrl100: string
  collectionPrice: number
  trackPrice: number
  collectionHdPrice: number
  releaseDate: string
  collectionExplicitness: string
  trackExplicitness: string
  trackCount: number
  trackTimeMillis: number
  country: string
  currency: string
  primaryGenreName: string
  contentAdvisoryRating: string
  artworkUrl600: string
  genreIds: string[]
  genres: string[]
}
