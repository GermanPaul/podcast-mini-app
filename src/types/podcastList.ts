export interface FetchPodcastFeedInput {
  limit: number
  genre: number
}

export interface PodcastFeed {
  id: string
  name: string
  author: string
  image: string
}
