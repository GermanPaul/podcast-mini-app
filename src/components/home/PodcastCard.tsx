import React from 'react'
import { Link } from 'react-router-dom';
import { PodcastFeed } from '@/types/podcastList'

interface HomeProps {
  podcast: PodcastFeed
}

const PodcastCard = ({ podcast }: HomeProps) => {
  return <Link to={`/podcast/${podcast.id}`} className='w-80 text-center m-2 shadow-none hover:shadow-lg'>
    <div className='block mx-auto rounded-full w-44'>
      <img className='mx-auto rounded-full -mb-20' height='170px' width='170px' src={podcast.image} alt={podcast.name} loading='lazy' />
    </div>
    <div className='block border border-t-0 px-2 pt-24 pb-5'>
      <div className='uppercase font-semibold'>{podcast.name}</div>
      <div>{podcast.author}</div>
    </div>
  </Link>
}

export default PodcastCard
