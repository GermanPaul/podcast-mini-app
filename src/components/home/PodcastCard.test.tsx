import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { PodcastFeed } from '@/types/podcastList'

import PodcastCard from './PodcastCard'

const podcastFeed: PodcastFeed = {
  id: '1',
  name: 'My first podcast',
  author: 'Rookie',
  image: 'https://image.provider/rookie/my-first-podcast.jpg'
}

describe('<PodcastCard />', () => {
  it('render the component', () => {
    const { baseElement } = render(<BrowserRouter>
      <PodcastCard podcast={podcastFeed} />
    </BrowserRouter>)

    expect(baseElement).toMatchSnapshot()
  })

  it('should implement a link to the podcast details page', () => {
    render(<BrowserRouter>
      <PodcastCard podcast={podcastFeed} />
    </BrowserRouter>)

    expect(screen.getByRole('link')).toHaveProperty('href', '/podcast/1')
  })

  it('should load the podcast image', () => {
    const { container } = render(<BrowserRouter>
      <PodcastCard podcast={podcastFeed} />
    </BrowserRouter>)

    expect(container.querySelector('img')).toHaveProperty('src', 'https://image.provider/rookie/my-first-podcast.jpg')
  })

  it('should display the podcast name', () => {
    render(<BrowserRouter>
      <PodcastCard podcast={podcastFeed} />
    </BrowserRouter>)

    expect(screen.getByTestId('podcast-name')).toHaveTextContent('My first podcast')
  })

  it('should display the podcast author', () => {
    render(<BrowserRouter>
      <PodcastCard podcast={podcastFeed} />
    </BrowserRouter>)

    expect(screen.getByTestId('podcast-author')).toHaveTextContent('Rookie')
  })
})
