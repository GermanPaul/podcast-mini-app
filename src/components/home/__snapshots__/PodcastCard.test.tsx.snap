// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PodcastCard /> > render the component 1`] = `
<body>
  <div>
    <a
      class="w-80 text-center m-2 shadow-none hover:shadow-lg"
      href="/podcast/1"
    >
      <div
        class="block mx-auto rounded-full w-44"
      >
        <img
          alt="My first podcast"
          class="mx-auto rounded-full -mb-20 w-44 h-44"
          loading="lazy"
          src="https://image.provider/rookie/my-first-podcast.jpg"
        />
      </div>
      <div
        class="block border border-t-0 px-2 pt-24 pb-5"
      >
        <div
          class="uppercase font-semibold"
          data-testid="podcast-name"
        >
          My first podcast
        </div>
        <div
          data-testid="podcast-author"
        >
          Rookie
        </div>
      </div>
    </a>
  </div>
</body>
`;
